File: cellulosesz-api/src/main/java/top/likoslupus/cellulosesz/api/economy/TransactionCause.java
package top.likoslupus.cellulosesz.api.economy;

public record TransactionCause(
        String type,
        String actor,
        String note
) {

    public static TransactionCause command(String actor, String note) {
        return new TransactionCause("command", actor, note);
    }

    public static TransactionCause system(String note) {
        return new TransactionCause("system", "system", note);
    }

}

File: cellulosesz-api/src/main/kotlin/top/likoslupus/cellulosesz/api/economy/TransactionCause.kt
package top.likoslupus.cellulosesz.api.economy

@JvmRecord
public data class TransactionCause(
    public val type: String,
    public val actor: String,
    public val note: String
) {

    public companion object {

        @JvmStatic
        public fun command(actor: String, note: String): TransactionCause =
            TransactionCause("command", actor, note)

        @JvmStatic
        public fun system(note: String): TransactionCause =
            TransactionCause("system", "system", note)

    }

}
